fastlane/Fastfile

```diff
@@ -238,12 +257,71 @@ end
       project_slug: 'simplenote-macos',
       dsym_path: dSYM_PATH,
     )
-   
+  end
+
+  lane :build_and_upload_app_store do | options |
+    # This allows code signing to work on CircleCI. It is skipped if this isn't
+    # running on CI.
+    #
+    # See https://circleci.com/docs/2.0/ios-codesigning/
+    setup_circle_ci
+
+    sync_code_signing(
+      type: 'appstore',
+      platform: 'macos',
+      # Will default to false, unless a readonly value is specified, or is
+      # running on CI, in which case it will always be true.
+      readonly: options[:readonly] || is_ci,
+      app_identifier: 'com.automattic.SimplenoteMac'
+    )
+
+    archive_path = File.join(@BUILD_FOLDER, 'Simplenote-Mac.xcarchive')
+    build_simplenote(codesign: true, archive_path: archive_path)
+
+    # TestFlight does not suport macOS but we can use deliver to upload the
+    # binary to the App Store by telling it to skip all the metadata,
+    # screenshots, etc.
+    #
+    # TODO: disabled while testing
+    #
+    # deliver(
+    #   skip_screenshots: true,
+    #   overwrite_screenshots: false,
+    #   skip_metadata: true,
+    #   skip_app_version_update: true,
+    #   skip_binary_upload: false,
+    #   force: true, # Skip verification of HTML preview file
+    # )
+
+    sentry_upload_dsym(
+      auth_token: get_required_env("SENTRY_AUTH_TOKEN"),
+      org_slug: 'a8c',
+      project_slug: 'simplenotemacos',
+      # At the time of writing, there's no way to explicitly configure the
+      # dSYM path, but build_mac_app sets it in the environment if successful.
+      # See `bundle exec fastlane action build_mac_app`.
+      dsym_path: ENV['DSYM_OUTPUT_PATH'],
+    )
+
+    # Do not create the GitHub release unless explicitly requested
+    next unless options[:create_github_release]
+
+    archive_zip_path = "#{archive_path}.zip"
+    zip(path: archive_path, output_path: archive_zip_path)
+
+    create_release(
+      repository: GHHELPER_REPO,
+      # version: ios_get_app_version(),
+      version: "test-#{Time.now.to_i}",
+      release_notes_file_path: 'Simplenote/Resources/release_notes.txt',
+      release_assets: [archive_zip_path],
+      prerelease: options[:prerelease],
+    )
```

Comment:
~This action should be called conditionally. At least that's what is suggested by the fact that you call the `build_and_upload_app_store` lane in `.circleci/config.yml` using `create_github_release:true`… that you don't seem to use here~

nvm, I missed that you had the `next unless options[:create_github_release]` above.

Usually I like guard statements to exit early, but having one here in the middle of the code (as opposed to at having guard clauses at the beginning of the function/lane, which is the usual place where we find/use them) makes me think they are harder to spot (proof: I almost missed it 😛 ) in such contexts (= mid-functions). wdyt of nesting the code inside a big `if` instead here?

Reply:
I prefer it like this because it's linear, but I do see your point.

Other projects use the `if` approach, too. Between that and your point, I'm happy to [disagree and commit](https://en.wikipedia.org/wiki/Disagree_and_commit) 👍 (That is, I'll update to use the `if`)